library/src/main/java/io/getstream/chat/android/client/events/ChatEvent.kt

```diff
@@ -1,48 +1,457 @@
 package io.getstream.chat.android.client.events
 
 import com.google.gson.annotations.SerializedName
+import io.getstream.chat.android.client.errors.ChatError
 import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.ChannelMute
 import io.getstream.chat.android.client.models.Member
 import io.getstream.chat.android.client.models.Message
 import io.getstream.chat.android.client.models.Reaction
 import io.getstream.chat.android.client.models.User
 import java.util.Date
 
-open class ChatEvent(val type: String = "") {
+sealed class ChatEvent {
+    abstract val type: String
+    abstract val createdAt: Date
+}
 
-    var cid: String? = null
+sealed class CidEvent : ChatEvent() {
```

Comment:
ChannelEvent would be a bit nicer name perhaps? @JcMinarro

Reply:
I was thinking about this name as well, but it could be a bit confused because `channel` instance is not present, only `cid`, `channelType` and `channelId`